packages/react/src/Details/Details.tsx

```diff
@@ -14,7 +18,82 @@ const Details = styled.details<SxProp>`
   ${sx};
 `
 
-Details.displayName = 'Details'
+export const StyledSummary = styled.summary<SxProp>`
+  ${sx};
+`
+
+const Root = React.forwardRef<HTMLDetailsElement, ComponentProps<typeof StyledDetails>>(
+  ({children, ...props}: ComponentProps<typeof StyledDetails>, forwardRef) => {
+    const detailsRef = React.useRef<HTMLDetailsElement>(null)
+    const ref = useMergedRefs(forwardRef, detailsRef)
+    const [hasSummary, setHasSummary] = useState(false)
+
+    useEffect(() => {
+      const {current: details} = detailsRef
+      if (!details) {
+        return
+      }
+
+      const updateSummary = () => {
+        const summary = details.querySelector('summary:not([data-default-summary])')
+        setHasSummary(!!summary)
+        warning(!summary, 'No summary found, a default summary will be rendered instead.')
```

Comment:
Sorry if I missed a response to this in a previous comment, couldn't find where I asked about this 🤔 

Is our thinking here that we want to allow folks to have a fallback summary or do we want to prevent folks from not including a summary? If it's the former, I don't know if we'd need a warning here. And if it's the latter I think we could just throw an error, right? Or am I misunderstanding? 👀

Reply:
No worries, I answered [here](https://github.com/primer/react/pull/5015#discussion_r1809488486). I think based on that I can remove it then :)

Reply:
removed!